src/main/java/net/logstash/logback/appender/AsyncDisruptorAppender.java

```diff
@@ -422,57 +450,127 @@ public void stop() {
         if (!super.isStarted()) {
             return;
         }
+        
         /*
          * Don't allow any more events to be appended.
          */
         super.stop();
-        try {
-            this.disruptor.shutdown(1, TimeUnit.MINUTES);
-        } catch (TimeoutException e) {
+
+        
+        /*
+         * Shutdown Disruptor
+         *
+         * Calling Disruptor#shutdown() will wait until all enqueued events are fully processed,
+         * but this waiting happens in a busy-spin. To avoid wasting CPU we wait for at most the configured
+         * grace period before asking the Disruptor for an immediate shutdown.
+         */
+        long deadline = getShutdownGracePeriod().getMilliseconds() < 0 ? Long.MAX_VALUE : System.currentTimeMillis() + getShutdownGracePeriod().getMilliseconds();
+        while (!isRingBufferEmpty() && (System.currentTimeMillis() < deadline)) {
+            LockSupport.parkNanos(SLEEP_TIME_DURING_SHUTDOWN);
+        }
+        
+        this.disruptor.halt();
```

Comment:
_nit_  extract into separate method

Reply:
I like to keep `start()` and `stop()` quite the same and the opposite of each other.
If I move the the code to stop the disruptor into a separate method, then I think we should do the same for the start... but I'm not sure it will add that much to the readability of the code.

Reply:
Ok, this just seemed like a nicely isolated piece of code that would be a good candidate to be extracted as a method.

It's essentially a re-implementation of disruptor.shutdown, which was previously called from stop.  So changing the stop method from calling this.disruptor.shutdown to this.shutdownDisruptor seemed rather nice.

Reply:
FYI I did not refactor the stop() method as you propose... It is likely to be affected by other up coming changes I have in mind ;-)